python/pyspark/sql/functions.py

```diff
@@ -33,17 +33,36 @@
 __all__ = ['countDistinct', 'approxCountDistinct', 'udf']
 
 
-def _create_function(name, doc=""):
+def _function_obj(sc, is_math=False):
+    if not is_math:
+        return sc._jvm.functions
+    else:
+        return sc._jvm.mathfunctions
+
+
+def _create_function(name, doc="", is_math=False, binary=False):
```

Comment:
not sure if you got my previous comment. it might be easier if is_math just takes a jvm object, rather than using an extra _function_obj